src/components/list/SearchResults.vue

```diff
@@ -0,0 +1,233 @@
+        && !this.isMobileScreen;
+    },
+    searchValue() {
+      return this.$route.query.searchValue;
+    },
+  },
+  watch: {
+    searchValue() {
+      this.setSearchResults(null);
+      this.searchVaultAndFindTextOnPage();
+    },
+  },
+  created() {
+    if (!this.lists?.length) {
+      this._fetchListsForUser();
+    }
+
+    const queryOptions = {
+      searchValue: {
+        callback: this.searchVaultAndFindTextOnPage,
+      },
+    };
+
+    handleQueryOnLoad(queryOptions, this.$route.query);
+    deleteExtraQuery(queryOptions, this.$route.query);
+  },
+  beforeUnmount() {
+    deleteFromQuery('searchValue');
+    this.setSearchResults(null);
+  },
+  methods: {
+    ...mapMutations([
+      'setSearchResults',
+    ]),
+    ...mapActions([
+      '_searchVault',
+    ]),
+    ...mapActions('lists', [
+      '_fetchListsForUser',
+      '_setListIdFromLocalStorage',
+    ]),
+    searchVaultAndFindTextOnPage() {
+      this._searchVault(this.searchValue)
+        .then(() => {
+          this.highlightResultText();
+        });
+    },
+    highlightResultText() {
+      const resultUnits = document.querySelectorAll('.result-unit');
+      const searchValueLowerCased = this.searchValue.toLowerCase();
+      const searchValueLength = this.searchValue.length;
+      const searchValueRegex = new RegExp(`${this.searchValue}`, 'ig');
+
+      resultUnits.forEach(unit => {
+        if (unit.innerText.toLowerCase().includes(searchValueLowerCased)) {
+          const initialHTML = unit.innerHTML;
+          const initialHTMLLowerCased = unit.innerHTML.toLowerCase();
+
+          const processedChunks = [];
+
+          let firstIndex = initialHTMLLowerCased.indexOf(searchValueLowerCased);
+          let lastIndex = firstIndex + searchValueLength;
+          let currentChunk = initialHTML.slice(0, lastIndex);
+
+          while (currentChunk.length >= searchValueLength) {
+            const originalCasedText = currentChunk.slice(firstIndex, lastIndex);
+            const newHTMLPiece = `<mark>${originalCasedText}</mark>`;
+            
+            const replacedChunk = currentChunk.replace(searchValueRegex, newHTMLPiece);
+
+            processedChunks.push(replacedChunk);
+
+            const extraSymbolsCount = (newHTMLPiece.length - originalCasedText.length)
+              * processedChunks.length;
+            const processedChunksLength = processedChunks.join('').length;
+            const newBeginningIndex = processedChunksLength - extraSymbolsCount;
+
+            const remainingPart = initialHTMLLowerCased.slice(newBeginningIndex);
+
+            firstIndex = remainingPart.indexOf(searchValueLowerCased);
+            lastIndex = firstIndex + searchValueLength;
+
+            currentChunk = initialHTML.slice(newBeginningIndex, newBeginningIndex + lastIndex);
+
+            if (firstIndex === -1) {
+              processedChunks.push(remainingPart);
+              currentChunk = '';
+            }
```

Comment:
this check is better to make exactly after calculating this index
and break this iteration